scenario/context.py

```diff
@@ -39,34 +40,51 @@
 
 @dataclasses.dataclass(frozen=True)
 class ActionOutput(_max_posargs(0)):
-    """Wraps the results of running an action event with ``run_action``."""
+    """Wraps the results of running an action event on a unit.
 
-    state: "State"
-    """The charm state after the action has been handled.
+    Tests should generally not create instances of this class directly, but
+    rather use the :attr:`Context.action_output` attribute to inspect the
+    results of running actions.
+    """
 
-    In most cases, actions are not expected to be affecting it."""
-    logs: List[str]
+    logs: List[str] = dataclasses.field(default_factory=list)
     """Any logs associated with the action output, set by the charm with
     :meth:`ops.ActionEvent.log`."""
+
     results: Optional[Dict[str, Any]] = None
     """Key-value mapping assigned by the charm as a result of the action.
     Will be None if the charm never calls :meth:`ops.ActionEvent.set_results`."""
-    failure: Optional[str] = None
-    """None if the action was successful, otherwise the message the charm set with
-    :meth:`ops.ActionEvent.fail`."""
 
-    @property
-    def success(self) -> bool:
-        """True if this action was a success, False otherwise."""
-        return self.failure is None
+    status: str = "pending"
```

Comment:
Why are these attributes and methods public, including the `set_*` methods?

Reply:
I missed making them private when I changed from private `_Task` to public `Task`. Will fix.